guides/common/modules/con_resolving-package-dependencies.adoc

```diff
@@ -3,16 +3,44 @@
+The more content and repositories in the content view, the greater the delay.
+
+The package dependency resolution feature does not take which packages are installed on your systems into account.
+Only packages within the content view's repositories are considered.
+For example, if a content view has only AppStream in it, dependency solving will not know to pull dependent BaseOS content in at publish time.
+See {ContentManagementDocURL}limitations-to-repository-dependency-resolution[Limitations to Repository Dependency Resolution] for more information about when dependency resolution may fail.
 
 .Resolving Package Dependencies and Filters
 
-Filters do not resolve any dependencies of the packages listed in the filters.
+Filters, by default, do not resolve any dependencies of the packages listed in the filters.
 This might require some level of testing to determine what dependencies are required.
 
 If you add a filter that excludes some packages that are required and the Content View has dependency resolution enabled, {Project} ignores the rules you create in your filter in favor of resolving the package dependency.
 
-If you create a content filter for security purposes, to resolve a package dependency, {Project} can add packages that you might consider insecure.
+If you create a content filter for security purposes but choose to resolve dependencies, {Project} can add packages that you might consider insecure.
+
+.Filtering and Dependency Solving Recommendations
+
+Be mindful when combining exclusion filters with dependency solving.
+For example, take a scenario where a user wants to recreate RHEL 8.3 via content view filters and include select packages and errata from a later RHEL 8 y-release.
+The user excludes all errata after the RHEL 8.3 release date minus some select few that are needed and enables dependency solving.
+Dependency solving may pull in more RPMs than expected, which would cause the system to stray further from being a RHEL 8.3 machine.
+If dependency solving pulls in too many unwanted RPMs, consider identifying by hand the core basic dependencies that the extra packages and errata need.
+
+If, in the above case, the user does not need the extra errata and RPMs, then content view filtering is not needed at all.
+The RHEL 8.3 repository could be enabled and used directly from the Red Hat Repositories page.
+
+In some cases, excluding RPMs may make dependency solving impossible.
+For example, if a user wants to make a RHEL 8.3 repo with a few RPMs missing, dependency solving should not be used because the missing RPMs could be dependencies for another RPMs in the repository.
+
+Dependency solving is most useful for cases like incremental update, where a critical security patch needs to be added to a content view version with all of its dependent packages.
```

Comment:
I'm not sure I agree with this statement.  Dependency solving is most useful for papering over problems caused by incorrect security metadata.  In nearly all cases, a critical security patch isn't going to require any dependency changes.

Reply:
@dralley if a user has an out-of-date RHEL 8 content view version and needs to add a new erratum, it's not possible that the erratum would require package updates outside of its package list?

Reply:
On the line below I have "While an erratum should include all related dependencies needed to apply a critical patch, there may be missing dependencies if content view filters are applied."

Do you feel that this isn't correct?

Reply:
I suppose if an erratum is an update for `packageA` and that package requires `packageB` and `packageC`, and it's applicable to a host, `packageA` must be in the host's content view version with its B and C dependencies already.

As in, there is no situation where content view filtering would not include B and C but include A and have A be installable.  And if A needs an update via an erratum, B and C will be included if they were updates as well.

Reply:
I think I may just take lines 40 and 41 out. I'll keep line 43 because that's package incremental update, which dependency solving could actually be useful for.

Reply:
>@dralley if a user has an out-of-date RHEL 8 content view version and needs to add a new erratum, it's not possible that the erratum would require package updates outside of its package list? And I'm asking aside from incorrect security metadata.

I'm saying it is possible, not even because the errata metadata is wrong but because the RPM metadata is wrong, and doesn't tell the dependency solver that it needs some version X.  If there was a mistake there (which would be really easy to make), nobody running up-to-date RHEL would necessarily ever notice, because getting the latest would paper over it.

I think in practice this probably wouldn't happen often unless you start cherrypicking feature errata.

>On the line below I have "While an erratum should include all related dependencies needed to apply a critical patch, there may be missing dependencies if content view filters are applied."

>Do you feel that this isn't correct?

It's partially correct.  The erratum only lists the packages that fix, or are built alongside the package that fixes the errata. The "dependencies" otherwise are just normal RPM dependencies, they aren't listed in the erratum.

>Edit: on second thought, this is assuming the content view version isn't super old -- what if the system is behind a few y-versions and a package gained some new dependencies? I suppose dependency solving might pull so many packages in from the Library repository that the user should likely just publish a new content view version.

Yeah, this goes back to cherrypicking feature errata being a bad idea.  I don't think security errata are likely to ever pull in entire new dependencies, but cherrypicking feature errata is tempting fate.  At that point you should really just update the CV.